drink.pyで検出されるpyright警告について

現在のコードだとpyrightでreportGeneralTypeIssuesが検出される。
一応明示的にstrとType Hintをつける事で回避できることがわかったが、これで良いのかよくわからない。
pyrightのバグなのか確認したいためIssueを建てました。（英語にツッコミどころが多い）

## log

https://github.com/AtsushiOzawa/tdd_vending-machine/runs/3482653202

```
+ pyright@1.1.165
added 1 package from 1 contributor and audited 1 package in 0.779s
found 0 vulnerabilities

Loading configuration file at /home/runner/work/tdd_vending-machine/tdd_vending-machine/pyrightconfig.json
No include entries specified; assuming /home/runner/work/tdd_vending-machine/tdd_vending-machine
Auto-excluding **/node_modules
Auto-excluding **/__pycache__
Auto-excluding .git
stubPath /home/runner/work/tdd_vending-machine/tdd_vending-machine/typings is not a valid directory.
Searching for source files
Found 10 source files
/home/runner/work/tdd_vending-machine/tdd_vending-machine/vending_machine/drink.py
  /home/runner/work/tdd_vending-machine/tdd_vending-machine/vending_machine/drink.py:31:12 - error: Expression of type "Literal['cola']" cannot be assigned to declared type "property"
    "Literal['cola']" is incompatible with "property" (reportGeneralTypeIssues)
  /home/runner/work/tdd_vending-machine/tdd_vending-machine/vending_machine/drink.py:39:12 - error: Expression of type "Literal['水']" cannot be assigned to declared type "property"
    "Literal['水']" is incompatible with "property" (reportGeneralTypeIssues)
  /home/runner/work/tdd_vending-machine/tdd_vending-machine/vending_machine/drink.py:47:12 - error: Expression of type "Literal['お茶']" cannot be assigned to declared type "property"
    "Literal['お茶']" is incompatible with "property" (reportGeneralTypeIssues)
3 errors, 0 warnings, 0 infos 
Completed in 1.706sec
Error: Process completed with exit code 1.
```

## 検出されるコード（最小コード）

```python
from abc import ABC, abstractmethod
class Drink(ABC):
    @property
    @abstractmethod
    def name(self) -> str:
        pass

class Cola(Drink):
    name = "cola"
```

## 回避するコード

```python
from abc import ABC, abstractmethod
class Drink(ABC):
    @property
    @abstractmethod
    def name(self) -> str:
        pass

class Cola(Drink):
    name: str = "cola"
```

## Issue
https://github.com/microsoft/pyright/issues/2253

--- a/vending_machine/drink.py
+++ b/vending_machine/drink.py
@@ -28,7 +28,7 @@ class Cola(Drink):
     コーラ
     """
 
-    name = "cola"
+    name: str = "cola"
 
 
 class Water(Drink):
@@ -36,7 +36,7 @@ class Water(Drink):
     水
     """
 
-    name = "水"
+    name: str = "水"
 
 
 class Tea(Drink):
@@ -44,4 +44,4 @@ class Tea(Drink):
     お茶
     """
 
-    name = "お茶"
+    name: str = "お茶"

